tests/test_stateful_browser.py

```diff
@@ -170,6 +170,39 @@ def test_submit_btnName(expected_post):
     assert initial_state != browser._StatefulBrowser__state
 
 
+@pytest.mark.parametrize("expected_post", [
+    pytest.param(
+        [
+            ('text', 'Setting some text!'),
+            ('comment', 'Selecting an input submit'),
+            ('preview', 'Preview Page'),
+            ('diff', 'Review Changes'),
+            ('save', 'Submit changes'),
+            ('cancel', 'Cancel'),
+        ], id='input'),
+    pytest.param(
+        [
+            ('text', '= Heading =\n\nNew page here!\n'),
+            ('comment', 'Selecting a button submit'),
+            ('preview', 'Preview Page'),
+            ('diff', 'Review Changes'),
+            ('save', 'Submit changes'),
+            ('cancel', 'Cancel'),
```

Comment:
We may need to rethink the implementation of this change, because it posts _all_ of the submit elements, rather than none of them:

```html
        <input type="submit" name="preview" value="Preview Page" />
        <input type="submit" name="diff" value="Review Changes" />
        <input type="submit" id="save" name="save" value="Submit changes" />
        <button type="submit" name="cancel" value="Cancel" />
```
I assume this is not the intended behavior? If we don't want any of the submit elements to be selected, then we probably need to modify `form.choose_submit` to delete all submit inputs if `submit is False` (or perhaps a new method with some refactoring).

Reply:
Yeah, I was wondering about that... Wasn't sure what the behaviour ought to be, but that's fair enough. I'll have a look into it.